camayoc/tests/qpc/ui/utils.py

```diff
@@ -199,6 +216,63 @@ def delete_credential(view, names):
                 xpath=row_xpath(name)), timeout=1)
 
 
+def edit_credential(view, original_name, options):
+    """Edit a credential through the UI and verify it was edited.
+
+    :param view: The view context (should be the browser view)
+    :param original_name: The original name of the credential.
+    :param options: The options to be edited within the credential.
+    """
+    view.refresh()
+    dash = DashboardView(view)
+    dash.nav.select('Credentials')
+    view.wait_for_element(locator=Locator(
+        xpath=(edit_xpath(original_name))))
+    GenericLocatorWidget(view, locator=Locator(
+        xpath=edit_xpath(original_name))).click()
+    modal = CredentialModalView(view, locator=Locator(css='.modal-content'))
+    if 'name' in options:
```

Comment:
All but "SSH Key File" have very close filling process. If you used the fields' label for the options keys then you could use a for loop and an if block to handle the exceptional case (SSH Key File).

Reply:
good idea, i'll look at doing this.